src/dct/templates/STM.lua

```diff
@@ -122,7 +122,7 @@ end
 --    }}}
 --]]
 
-function STM.transform(stmdata, file)
```

Comment:
I think you still need to keep the `file` function parameter or edit src/dct/templates/Template.lua +520 to not pass the file name. The file name is provided by the utils.readlua call. Also, it would seem useful to keep the filename name for error reporting which is what originally it was used for.

Reply:
As the `file` parameter is not used here anymore (the coalition check was moved to the Template validation), one alternative could be to set it as a field of Template.

Reply:
Ok my only point was where `STM.transform` is used when called with `readlua`, `STM.transform(utils.readlua(...))`, readlua returns a two tuple of stmdata and the file path.